src/terrain/routes/token.clj

```diff
@@ -14,4 +14,36 @@
        :summary "Obtain OAuth Tokens"
        :return AccessTokenResponse
        :description-file "docs/get-token.md"
-       (get-token authorization)))))
+       (oauth/get-token authorization))
+
+     (GET "/cas" [:as {{:strs [authorization]} :headers}]
+       :summary "Obtain OAuth Tokens"
+       :return AccessTokenResponse
+       :description-file "docs/get-token.md"
+       (oauth/get-cas-token authorization))
+
+     (GET "/keycloak" [:as {{:strs [authorization]} :headers}]
+       :summary "Obtain Keycloak OIDC Tokens"
+       :return AccessTokenResponse
+       :description-file "docs/get-token.md"
+       (oauth/get-keycloak-token authorization)))))
+
+(defn admin-token-routes
+  []
+  (routes
+   (context "/admin/token" []
+     :tags ["admin-token"]
+
+     (GET "/" [:as {{:strs [authorization]} :headers}]
+       :query [{:keys [username]} AdminKeycloakTokenParams]
+       :summary "Obtain Impersonation Tokens"
+       :return AccessTokenResponse
+       :description-file "docs/get-admin-token.md"
+       (oauth/get-admin-token authorization username))
+
+     (GET "/keycloak" [:as {{:strs [authorization]} :headers}]
+       :query [{:keys [username]} AdminKeycloakTokenParams]
+       :summary "Obtain Keycloak OIDC Impersonation Tokens"
+       :return AccessTokenResponse
+       :description-file "docs/get-admin-token.md"
+       (oauth/get-keycloak-admin-token authorization username)))))
```

Comment:
For the time being, we also have a default admin token endpoint as well as an endpoint for the one identity provider that we support that allows us to get impersonation tokens.

Also the admin token endpoints use HTTP basic authentication as well. Because of the way that Keycloak allows users to obtain impersonation tokens, this means that these endpoints have to make two separate API calls: one to get the administrator's token, and another to get the actual impersonation token. I could have made the admin token endpoints require a bearer token instead, but that makes it a bit more cumbersome to use these endpoints. I also could have made the endpoints support both HTTP basic authentication and bearer token authentication, but that would have taken longer, and it's not worth the additional time and effort at the moment.